src/simulator/views/map/object/square_mesh.py

```diff
@@ -81,15 +73,15 @@ def make(x1, y1, z1, x2, y2, z2, use_tex: bool = False) -> None:
 
             self.__face_count += 1
 
-        if depth == 0:
-            make(0, 0, 0, width, height, 0)
+        if self.depth == 0:
+            make(0, 0, 0, width, height, 0, [(1.0, 1.0), (0.0, 1.0), (0.0, 0.0), (1.0, 0.0)])
         else:
-            make(width, height, -depth, 0, height, 0)
-            make(0, 0, -depth, width, 0, 0)
-            make(width, 0, -depth, width, height, 0)
-            make(0, height, -depth, 0, 0, 0)
-            make(width, height, 0, 0, 0, 0, True)
-            make(0, height, -depth, width, 0, -depth)
+            make(0, height, -self.depth, 0, 0, 0, ((0.0, 1.0), (0.0, 0.0), (0.0, 0.0), (0.0, 1.0))) # SIDE
+            make(0, 0, -self.depth, width, 0, 0, ((0.0, 0.0), (1.0, 0.0), (1.0, 0.0), (0.0, 0.0)))  # SIDE
+            make(width, height, -self.depth, 0, height, 0, ((1.0, 1.0), (0.0, 1.0), (0.0, 1.0), (1.0, 1.0))) # SIDE
```

Comment:
good comments

Reply:
Yh I don't even know how it works. Just trial and error